word_language_model/lstm_bucketing_infer.py

```diff
@@ -0,0 +1,153 @@
+                               shape=shape,
+                               layout=layout)])
+
+
+
+
+def tokenize_text(fname, vocab=None, invalid_label=-1, start_label=0):
+    try:
+        assert os.path.exists(fname)
+    except AssertionError:
+        subprocess.call(['{}/get_ptb_data.sh'.format(os.path.dirname(__file__))])
+        assert os.path.exists(fname)
+    lines = open(fname).readlines()
+    lines = [filter(None, i.split(' ')) for i in lines]
+    sentences, vocab = mx.rnn.encode_sentences(lines, vocab=vocab, invalid_label=invalid_label,
+                                               start_label=start_label)
+    return sentences, vocab
+
+def sym_gen(seq_len):
+    data = mx.sym.Variable("data")
+    embed = mx.sym.Embedding(data=data, input_dim=len(vocab),
+                             output_dim=num_embed, name="embed")
+    stack.reset()
+    outputs, _ = stack.unroll(seq_len, inputs=embed, merge_outputs=True)
+
+    pred = mx.sym.Reshape(outputs, shape=(-1, num_hidden))
+    pred = mx.sym.FullyConnected(data=pred, num_hidden=len(vocab), name="pred")
+    pred = mx.sym.softmax(pred, name='softmax')
+    return pred, ('data',), ('softmax_label',)
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+
+    model_dir = "model"
+
+    buckets = [10, 20, 30, 40, 50, 60]
+    start_label = 1
+    invalid_key = "\n"
+    invalid_label = 0
+    vocab = {}
+    idx2word = {}
+    epoch = 100
+
+    mx.test_utils.download("https://s3.amazonaws.com/model-server/models/lstm_ptb/lstm_ptb-symbol.json", dirname=model_dir)
+    mx.test_utils.download("https://s3.amazonaws.com/model-server/models/lstm_ptb/lstm_ptb-0100.params", dirname=model_dir)
+    mx.test_utils.download("https://s3.amazonaws.com/model-server/models/lstm_ptb/vocab_dict.txt", dirname=model_dir)
+    mx.test_utils.download("https://s3.amazonaws.com/model-server/models/lstm_ptb/signature.json", dirname=model_dir)
+
+    signature_file_path = os.path.join(model_dir, "signature.json")
+    if not os.path.isfile(signature_file_path):
+        raise RuntimeError("Missing signature.json file.")
+
+    with open(signature_file_path) as f:
+        signature = json.load(f)
+
+    data_names = []
+    data_shapes = []
+
+    for input_data in signature["inputs"]:
+        data_names.append(input_data["data_name"])
+        data_shapes.append((input_data["data_name"], tuple(input_data["data_shape"])))
+
+    vocab_dict_file = os.path.join(model_dir, "vocab_dict.txt")
+
+    with open(vocab_dict_file, 'r') as vocab_file:
+        vocab[invalid_key] = invalid_label
+        for line in vocab_file:
+            word_index = line.split(' ')
+            if len(word_index) < 2 or word_index[0] == '':
+                continue
+            vocab[word_index[0]] = int(word_index[1].rstrip())
+    for key, val in vocab.items():
+        idx2word[val] = key
+
+    # Load pre-trained lstm bucketing module
+    num_layers = 2
+    num_hidden = 200
+    num_embed = 200
```

Comment:
All training benchmarks use 650 hidden and 650 embedded nodes. Any particular reason why you are using 200 nodes?

Reply:
I am using the pretrained MMS model which uses num_hidden and num_embed as 200. https://s3.amazonaws.com/model-server/models/lstm_ptb/lstm_ptb-symbol.json